apps/managedidentity/managedidentity_test.go

```diff
@@ -48,122 +75,178 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
 }
 
 func makeResponseWithErrorData(err string, desc string) ([]byte, error) {
-	responseBody := ErrorRespone{
+	responseBody := ErrorResponse{
 		Err:  err,
 		Desc: desc,
 	}
 	jsonResponse, e := json.Marshal(responseBody)
 	return jsonResponse, e
 }
 
-type resourceTestData struct {
-	source   Source
-	endpoint string
-	resource string
-	miType   ID
+func createMockFile(t *testing.T, path string, size int64) {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		t.Fatalf("failed to create directory: %v", err)
+	}
+	f, err := os.Create(path)
+	if err != nil {
+		t.Fatalf("failed to create file: %v", err)
+	}
+	if size > 0 {
+		if err := f.Truncate(size); err != nil {
+			t.Fatalf("failed to truncate file: %v", err)
+		}
+	}
+	f.Close()
```

Comment:
As written, this file handle leaks when `f.Truncate` returns an error. You need to `defer` this after `os.Create`